labcontrol/db/process.py

```diff
@@ -2520,35 +2520,75 @@ def _format_picklist(vol_sample, max_vol_per_well=60000,
         dest_plate_shape: list of 2 elements
             The destination plate shape
         """
+
+        def get_well_name(offset_from_ascii_ucase_a, col_num):
+            row_ascii_code = ord('A') + offset_from_ascii_ucase_a
+            row_letter = chr(row_ascii_code)
+            return "%s%d" % (row_letter, col_num)
+
         if dest_plate_shape is None:
             dest_plate_shape = [16, 24]
 
         contents = ['Source Plate Name,Source Plate Type,Source Well,'
                     'Concentration,Transfer Volume,Destination Plate Name,'
                     'Destination Well']
-        # Write the sample transfer volumes
-        rows, cols = vol_sample.shape
+        num_input_rows, num_input_cols = vol_sample.shape
+        running_tot = 0
+        num_dest_rows = dest_plate_shape[0]
+        num_dest_cols = dest_plate_shape[1]
+        dest_well_index = 0
+
         # replace NaN values with 0s to leave a trail of unpooled wells
         pool_vols = np.nan_to_num(vol_sample)
-        running_tot = 0
-        d = 1
-        for i in range(rows):
-            for j in range(cols):
-                well_name = "%s%d" % (chr(ord('A') + i), j + 1)
-                # Machine will round, so just give it enough info to do the
-                # correct rounding.
-                val = "%.2f" % pool_vols[i][j]
-                # test to see if we will exceed total vol per well
-                if running_tot + pool_vols[i][j] > max_vol_per_well:
-                    d += 1
-                    running_tot = pool_vols[i][j]
+
+        for curr_input_row_index in range(num_input_rows):
```

Comment:
I believe that weirdness would result if `num_input_rows` is `>= 26`. Similar to the [comment](https://github.com/jdereus/labman/pull/502/files#r282702728) in #502, should there be defensive bounds checking (...or is it somewhere deeper in the codebase...)?

Reply:
Oooo, great point--I like the cut of your edge-case jib!  However, we should be careful about where we add this check because there is nothing that inherently prevents a plate from having >26 rows (e.g., 1584-well plates have 32 rows) or that prevents the majority of this code from handling it.  The specific issue would, I believe, only be that the (newly encapsulated :D ) code here for generating the well NAMES wouldn't work with >26 rows and would need to be refactored accordingly (e.g., 1584s roll over to AA, AB, etc, like excel columns). I will therefore add a check in that sub-method that throws a ValueError if the offset is >25.  

That said, there is code to either generate or parse well names in various other places in LabControl too -- e.g., in the generate_prep_information methods where you caught this, and at least in the two additional locations below, and maybe more:

https://github.com/jdereus/labman/blob/645c798e0094e9962973c2d111c4860b1e8e5b23/labcontrol/db/container.py#L269-L279

https://github.com/jdereus/labman/blob/7bb9106555143bb4877c49516e92c792ea9349a5/labcontrol/db/process.py#L1828-L1861

Based on this, I suggest we also make an issue to do a sweep for affected code, because it definitely isn't all centralized ...

Reply:
:+1:. Excellent point with the >26 case and the other instances. Agree that a sweep issue makes sense -- thank you for the additional investigative work!